core/src/main/java/brooklyn/util/task/BasicExecutionContext.java

```diff
@@ -169,6 +169,11 @@ public T call() {
 
         taskTags.addAll(tags);
         
+        if (Tasks.current()!=null && BrooklynTaskTags.isTransient(Tasks.current()) && !taskTags.contains(BrooklynTaskTags.NON_TRANSIENT_TASK_TAG)) {
```

Comment:
Check for `!taskTags.contains(TRANSIENT_TASK_TAG)` to prevent multiple entries (in case already set by the user).